gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -241,10 +233,15 @@ public void orchestrate(Spec spec) throws Exception {
       if (!canRun(flowName, flowGroup, allowConcurrentExecution)) {
         _log.warn("Another instance of flowGroup: {}, flowName: {} running; Skipping flow execution since "
             + "concurrent executions are disabled for this flow.", flowGroup, flowName);
-        if (this.flowAlreadyRunningGauge.isPresent()) {
-          this.jobAlreadyRunning.incrementAndGet();
-        }
+        // We send a gauge with value 0 signifying that the flow could not be compiled because previous execution is already running
+        metricContext.newContextAwareGauge(
+            MetricRegistry.name(MetricReportUtils.GOBBLIN_SERVICE_METRICS_PREFIX, flowGroup, flowName, ServiceMetricNames.COMPILED),
+            () -> 0L);
```

Comment:
Can we use gauge with values 1 and 2 instead of 0 and 1?

Reply:
No, O and 1 seems more intuitive. 0 means not compiled, 1 means compiled?